tensorflow/c/eager/c_api.cc

```diff
@@ -460,17 +530,30 @@ TFE_Op* TFE_NewOp(TFE_Context* ctx, const char* op_or_function_name,
   const char* name = op_or_function_name;  // Shorthand
   const tensorflow::AttrTypeMap* types;
   status->status = tensorflow::AttrTypeMapForOp(name, &types);
-  if (status->status.ok()) return new TFE_Op(ctx, name, types);
+  if (status->status.ok()) {
+    if (ctx->input_attrs_inference) {
+      const tensorflow::OpDef* op_def;
+      status->status = tensorflow::OpDefForOp(op_or_function_name, &op_def);
+      TF_RETURN_IF_ERROR(status->status);
+      return new TFE_Op(ctx, name, types, new TFE_OpInferenceContext(op_def));
+    }
+    return new TFE_Op(ctx, name, types, nullptr);
+  }
   if (TF_GetCode(status) == TF_NOT_FOUND) {
     if (ctx->context.FindFunctionByName(name)) {
       status->status = tensorflow::Status::OK();
-      return new TFE_Op(ctx, name, nullptr);
+      return new TFE_Op(ctx, name, nullptr, nullptr);
     }
   }
   return nullptr;
 }
 
-void TFE_DeleteOp(TFE_Op* op) { delete op; }
+void TFE_DeleteOp(TFE_Op* op) {
+  if (op->inference_ctx != nullptr) {
```

Comment:
This looks like it could be a unique_ptr member instead of relying on the destructor for correctness.

Reply:
If we remove the option, then their won't be an `inference_ctx` and its data will be embedded within `TFE_Op`

Reply:
Since the `TFE_Op` has to be explicitly deleted itself by calling `TFE_DeleteOp`, I think it is more acceptable to delete its inference context explicitly as well.

Reply:
The TFE_Op has no destructor because we have a C API, but inside the C API we can use all of C++ and avoid risking memory bugs with future edits to this code. Please make it a unique_ptr.

Reply:
I see your point, I'll update it.